handlers/services.go

```diff
@@ -0,0 +1,23 @@
+package handlers
+
+import (
+	"net/http"
+
+	"github.com/gorilla/mux"
+	"github.com/swift-sunshine/swscore/log"
+	"github.com/swift-sunshine/swscore/models"
+)
+
+func ServiceList(w http.ResponseWriter, r *http.Request) {
+	params := mux.Vars(r)
+	namespace := models.Namespace{params["id"]}
+
+	services, err := models.GetServicesByNamespace(namespace.Name)
```

Comment:
is namespace's name and id different? why is this fetching done by name? just making sure there is a good reason for how it is done, not necessarily a problem

Reply:
@abonas  yes, name is the only way you can identify a namespace thought the K8s Client Go.

Here the definitions of the List methods within `k8s/client-go`:
[k8s/client-go/namespace.go](https://github.com/kubernetes/client-go/blob/7e03948729e653c2fdd124f1f3cd6f37b5a9c117/kubernetes/typed/core/v1/namespace.go#L65)
[k8s/client-go/service.go](https://github.com/kubernetes/client-go/blob/7e03948729e653c2fdd124f1f3cd6f37b5a9c117/kubernetes/typed/core/v1/service.go#L66)

Reply:
so why on line 13 params[id] is used and not name? or it's a different thing?

Reply:
yes, it is a different thing. `params["id"]` comes from the route, from gorilla/mux.

https://github.com/swift-sunshine/swscore/pull/51/files#diff-c270cd65cbdc0b86935559ad61065387R33

IMO, within the route `/namespaces/{id}/services` *id* is actually an identification literal.
Probably I am so influenced from Rails which it always used `id` (for single route like `namespaces/{id}) and `resource_id` (for nested routes like `/namespaces/{namespace_id}/services/{id}`

Reply:
Fwiw, I agree with @abonas, it's not really an id, it is just the the namespace name.  {id} is too generic, {namespace_id} is better but implies the id is different,  so I'd probably like to see: /api/namespaces/{namespace}/...

Reply:
+1 @jshaughn 
It will make it consistent with `/api/namespaces/{namespace}/services/{service}` that will come sooner or later. The fact that they are _identifiers_ is quite implicit while accessing resources.

Reply:
+1 using {namespace} instead of {id} is clearer.

Reply:
Changed! Gotta get rid of railsy way :sweat_smile: